src/styling-panel-definition.js

```diff
@@ -77,6 +80,191 @@ export const getStyleEditorDefinition = () => ({
+            colorPicker: {
+              type: 'object',
+              component: 'color-picker',
+              ref: 'style.background.color',
+              translation: 'properties.color.used',
+              disableNone: false,
+              defaultValue: styleDefaults.COLOR,
+              dualOutput: true,
+              show: (data) => !propertyResolver.getValue(data, 'style.background.useColorExpression'),
+            },
+          },
+        },
+        backgroundImage: {
+          items: {
+            backgroundImageMode: {
+              component: 'dropdown',
+              ref: 'style.background.mode',
+              translation: 'properties.backgroundImage',
+              defaultValue: styleDefaults.BGIMAGE_MODE,
+              options: [
+                {
+                  value: 'none',
+                  translation: 'Background.None',
+                },
+                {
+                  value: 'media',
+                  translation: 'MediaLibrary.Header',
+                },
+              ],
+              change(data) {
+                const bgImageComp = data.style.background;
+                bgImageComp
+                  ? (bgImageComp.url = { qStaticContentUrlDef: '' })
+                  : (data.style.background = { url: { qStaticContentUrlDef: '' } });
+              },
+            },
+            MediaLibrary: {
+              component: 'media-library-button',
+              ref: 'style.background.url',
+              translation: 'MediaLibrary.Header',
+              show: (data) => propertyResolver.getValue(data, 'style.background.mode') === 'media',
+            },
+            imageSize: {
+              component: 'dropdown',
+              ref: 'style.background.size',
+              defaultValue: styleDefaults.BACKGROUND_SIZE,
+              options: [
+                {
+                  value: 'auto',
+                  translation: 'properties.backgroundImage.originalSize',
+                },
+                {
+                  value: 'alwaysFit',
+                  translation: 'properties.backgroundImage.sizeAlwaysFit',
+                },
+                {
+                  value: 'fitWidth',
+                  translation: 'properties.backgroundImage.sizeFitWidth',
+                },
+                {
+                  value: 'fitHeight',
+                  translation: 'properties.backgroundImage.sizeFitHeight',
+                },
+                {
+                  value: 'stretchFit',
+                  translation: 'properties.backgroundImage.sizeStretch',
+                },
+                {
+                  value: 'alwaysFill',
+                  translation: 'properties.backgroundImage.sizeAlwaysFill',
+                },
+              ],
+              change: (data) => {
+                let currentPosition = propertyResolver.getValue(data, 'style.background.position');
+                if (currentPosition) currentPosition = styleDefaults.BGIMAGE_POSITION;
```

Comment:
Do we need to reset the position every time we change the size?

Reply:
Changing happens when you select another size mode and each option enables/disables some positions spots. Thus, yes it puts it back to center and let you choose the new position.